virtool/users/data.py

```diff
@@ -292,91 +354,120 @@ async def update(self, user_id: str, data: UpdateUserRequest):
-            )
+                if data["administrator"]:
+                    await self._authorization_client.add(role_assignment)
+                else:
+                    await self._authorization_client.remove(role_assignment)
 
-        if "password" in data:
-            update.update(
-                {
-                    "password": virtool.users.utils.hash_password(data["password"]),
-                    "last_password_change": virtool.utils.timestamp(),
-                    "invalidate_sessions": True,
-                }
-            )
+            if "force_reset" in data:
+                updates.update(
+                    {"force_reset": data["force_reset"], "invalidate_sessions": True}
+                )
+                user.force_reset = data["force_reset"]
+                user.invalidate_sessions = True
+
+            if "password" in data:
+                hashedpass = virtool.users.utils.hash_password(data["password"])
+                timestamp = virtool.utils.timestamp()
+                updates.update(
+                    {
+                        "password": hashedpass,
+                        "last_password_change": timestamp,
+                        "invalidate_sessions": True,
+                    }
+                )
+                user.password = hashedpass
+                user.last_password_change = timestamp
+                user.invalidate_sessions = True
+
+            if "groups" in data:
+                try:
+                    updates.update(
+                        await compose_groups_update(self._pg, data["groups"])
+                    )
+                    for _id in data["groups"]:
+                        await pg.execute(
+                            insert(user_group_associations).values(
+                                user_id=user.id, group_id=_id
+                            )
+                        )
 
-        if "groups" in data:
-            try:
-                update.update(await compose_groups_update(self._pg, data["groups"]))
-            except DatabaseError as err:
-                raise ResourceConflictError(str(err))
+                except DatabaseError as err:
+                    raise ResourceConflictError(str(err))
 
-        if "primary_group" in data:
-            try:
-                update.update(
-                    await compose_primary_group_update(
+            if "primary_group" in data:
+                try:
+                    primary_group = await compose_primary_group_update(
                         self._mongo,
                         self._pg,
                         data.get("groups", []),
                         data["primary_group"],
                         user_id,
                     )
+                    await pg.execute(
+                        insert(user_group_associations).values(
+                            user_id=user.id, group_id=data["primary_group"]
+                        )
```

Comment:
What happens to the users previous primary group?

Reply:
updated and now getting removed from table